code/modules/projectiles/ammunition/_firing.dm

```diff
@@ -21,11 +21,19 @@
 		next_delay = round(next_delay * 0.5)
 
 	user.changeNext_move(next_delay)
-
-	user.newtonian_move(get_dir(target, user))
+	if(!tk_firing(user, fired_from))
+		user.newtonian_move(get_dir(target, user))
+	else if(ismovable(fired_from))
+		var/atom/movable/firer = fired_from
+		if(!firer.newtonian_move(get_dir(target, fired_from), instant = TRUE))
+			var/throwtarget = get_step(fired_from, get_dir(target, fired_from))
+			firer.safe_throw_at(throwtarget, 1, 2)
```

Comment:
What does this case cover

Reply:
I don't understand the question

Reply:
If we can't newtonian move, we're not in space, so we throw ourselves backwards a tile to simulate recoil instead

Reply:
what is fired_from in this case?

Reply:
fired_from is the atom that fired the bullet

Reply:
aight, thanks